### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by dchest**
* Fixed `assert.throw` tests, which were catching wrong exceptions.
* Removed extra arguments passed to `redSqrt`.

### CODE REVIEW

File: test/constructor-test.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ describe('BN.js/Constructor', function () {
       var num = Math.pow(2, 53);
 
       assert.throws(function () {
-        BN(num, 10);
+        num = new BN(num, 10);
```
Comment (axic) [inline test/constructor-test.js]: Is this needed?

File: test/arithmetic-test.js [hunk 2]
```diff
@@ -599,12 +599,12 @@ describe('BN.js/Arithmetic', function () {
     });
     it('should not allow 0 input', function () {
       assert.throws(function () {
-        BN(1).egcd(0);
+        new BN(1).egcd(0);
```
Comment (axic) [inline test/arithmetic-test.js]: Good catch! Maybe it would be better checking for the actual exception message to fully verify what caused the throw?

Comment (fanatid) [inline test/arithmetic-test.js]: @dchest please add `/^Error: Assertion failed$/`

File: test/constructor-test.js (hunk 1 above)
Comment (dchest) [inline test/constructor-test.js]: `num =` ? Yes, otherwise pre-commit check complains about using `new` for side-effects.

**Comment by fanatid** [comment]
It's strange that `semistandard` say nothing about `BN` without `new`.

File: test/constructor-test.js (hunk 1 above)
Comment (fanatid) [inline test/constructor-test.js]: @dchest you can write `return` instead `num`

Comment (axic) [inline test/constructor-test.js]: And it doesn't complain on the other ones where no assignment is made?

Comment (dchest) [inline test/constructor-test.js]: OK, will do. No, others have other methods in chain, so it doesn't complain.

Comment (axic) [inline test/constructor-test.js]: Oh I see!

**Comment by fanatid** [comment]
@dchest can you also add second argument for other `assert.throws`? :) (my grep expression found 9 matches)

File: test/utils-test.js [hunk 3]
```diff
@@ -202,7 +202,7 @@ describe('BN.js/Utils', function () {
       assert.equal(new BN(0x3fffffe).cmpn(0x3fffffd), 1);
       assert.throws(function () {
         new BN(0x3fffffe).cmpn(0x4000000);
```
Comment (fanatid) [inline test/utils-test.js]: also need `return`

Comment (dchest) [inline test/utils-test.js]: It works because there's `cmpn` after `new`, so there's side-effect-only constructor, which `semistandard` reports about.

Comment (fanatid) [inline test/utils-test.js]: thanks for explanation!

**Comment by fanatid** [comment]
LGTM

**Comment by indutny** [comment]
Landed in c954522, thank you!

**Pull request closed without merging**